src/conda_devenv/devenv.py

```diff
@@ -688,72 +825,215 @@ def resolve_env_manager(args: argparse.Namespace) -> Literal["conda", "mamba"]:
+
+    if args.lock or args.update_locks:
+        return create_update_lock_file(env_manager, args)
 
-        # render conda-devenv file
-        conda_yaml_dict, environment = load_yaml_dict(filename)
-        rendered_contents = render_for_conda_env(conda_yaml_dict)
+    elif args.use_locks in ("auto", "yes"):
+        try:
+            return create_update_env_using_lock_file(env_manager, args)
+        except LockFileNotFoundError as e:
+            if args.use_locks == "yes":
+                raise UsageError(
+                    f"lock file {e.lock_file} not found and --use-locks=yes, aborting."
+                )
+            return create_update_env(env_manager, args)
+
+    else:
+        return create_update_env(env_manager, args)
 
-        if args.print or args.print_full:
-            print(rendered_contents)
-            if args.print_full:
-                print(render_for_conda_env({"environment": environment}, header=""))
-            return 0
 
-        # Write to the output file
+def print_rendered_environment(args: argparse.Namespace) -> int:
+    """
+    If print_full is True, includes the 'environment' section (for environment
+    variable definition), which is usually left out of the processed environment.yml
+    file.
+    """
+    render = ProcessedEnvironment.from_file(resolve_source_file(args))
+    print(render.rendered_yaml)
+    if (
+        args.print_full
+        and render.conda_yaml_dict
+        and "environment" in render.conda_yaml_dict
+    ):
+        print(
+            render_for_conda_env(
+                {"environment": render.conda_yaml_dict["environment"]}, header=""
+            )
+        )
+    return 0
+
+
+def create_update_env(
+    env_manager: Literal["conda", "mamba"], args: argparse.Namespace
+) -> int:
+    filename = resolve_source_file(args)
+    processed = ProcessedEnvironment.from_file(filename)
+    if processed.is_devenv_file:
+        # Write the processed contents into the equivalent environment.yml file.
         output_filename = __write_conda_environment_file(
-            args, filename, rendered_contents
+            args, filename, processed.rendered_yaml
         )
     else:
-        conda_yaml_dict = None
-        environment = {}
-        # Just call conda-env directly in plain environment.yml files
+        # Just call conda-env directly in plain environment.yml files.
         output_filename = filename
-        if args.print:
-            with open(filename) as f:
-                print(f.read())
-            return 0
 
-    env_name = get_env_name(args, output_filename, conda_yaml_dict)
+    # Hack around --prune not working correctly (at least in conda; mamba seems to work correctly).
+    env_name = get_env_name(args, processed.conda_yaml_dict)
     env_directory = get_env_directory(env_manager, env_name)
-
     if not args.no_prune:
         # Truncate the history file
         truncate_history_file(env_directory)
```

Comment:
Hmm not sure I follow, are you suggesting anything?

Reply:
Yes, removing this entirely.

Or checking if this hack is working again (I will not expect this).

Reply:
I will just drop it and mention it in the CHANGELOG then. 👍

Reply:
Dropped in 3d4a1d8

Reply:
AFAIK, the bug is still present in `conda`, so it could be a breaking change for `conda-devenv` when used together with `conda`. See https://github.com/conda/conda/issues/7279 (still open) and this PR that never got merged: https://github.com/conda/conda/pull/9614.

Reply:
> Last time I checked (a few years ago) this hack was worst than not prunning with "conda".

Reply:
This hack is currently doing harm and not fixing something.
If you intend to keep this we will very well restrict the conda version used to `< 4.7` (this hack stopped working in 4.7, I don't know if it is working again) but I don't think we want to do that.